src/utils/embedded/embedded.utils.ts

```diff
@@ -128,16 +130,36 @@ const {
 // type TRPCClient = ReturnType<typeof createTRPCProxyClient<AppRouter>>;
 // const trpcVanilla = client as TRPCClient;
 
-// Create a singleton instance of `SupabaseClient`
+// Create a singleton instance of `SupabaseClient` & `LocalStorage`
 let supabaseInstance: ReturnType<typeof createSupabaseClient> | null = null;
+let storage: LocalStorage | null = null;
 
-function getSupabaseClient() {
+export async function getStorage() {
+  if (!storage) {
+    storage = new LocalStorage();
+    await storage.requestStorageAccess();
+  }
+
+  return storage;
+}
+
+export async function getSupabaseClient() {
```

Comment:
As we should just use the `getSupabaseClient` function, let's remove the `supabase` export below (collapsed in the PR). Let's keep the call to `getSupabseClient()` though to make sure we initialize this as soon as possible.